samcli/lib/config/samconfig.py

```diff
@@ -106,7 +106,16 @@ def put(self, cmd_names, section, key, value, env=DEFAULT_ENV):
         if not self.document:
             self._read()
             # Empty document prepare the initial structure.
-            self.document.update({env: {self._to_key(cmd_names): {section: {key: value}}}})
+            # self.document is a nested dict, we need to check each layer and add new tables, otherwise duplicated key
+            # in parent layer will override the whole child layer
+            if self.document.get(env, None):
```

Comment:
is there a testcase to showcase all the cases shown here?

Reply:
Yes, they are tested in test_samconfig.py.